### PR TITLE
Add explainer replicas and handle scale to zero

### LINKED ISSUE #3241: explainer don't repect the spec.replicas
## Describe the bug
explainer don't repect the spec.replicas 

## To reproduce
1. deploy the model
```yaml 
apiVersion: machinelearning.seldon.io/v1
kind: SeldonDeployment
metadata:
  name: income
  namespace: dscoe
spec:
  name: income
  replicas: 1
  predictors:
  - graph:
      children: []
      implementation: SKLEARN_SERVER
      modelUri: gs://seldon-models/sklearn/income/model-0.23.2
      name: classifier
    explainer:
      type: AnchorTabular
      modelUri: gs://seldon-models/sklearn/income/explainer-py36-0.5.2
    name: default
```
2. update the income seldondeployment's spec.replicas = 0

## Current behaviour
Only classifer pods are terminated, explainer pods still running.

## Expected behaviour
Both classifer and explainer pods are terminated.

### PR DESCRIPTION
**Pull request by cliveseldon**
**What this PR does / why we need it**:

 * Scales explainer to zero if main predictor replicas is zero
 * Allows explainer replicas to be add. New addition to CRD.

**Which issue(s) this PR fixes**:

Fixes #3241

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **cliveseldon**
You can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: operator/controllers/seldondeployment_explainers_test.go [hunk 1]
```diff
@@ -156,6 +156,7 @@ var _ = Describe("Create a Seldon Deployment with explainer", func() {
 	const timeout = time.Second * 30
 	const interval = time.Second * 1
 	namespaceName := rand.String(10)
+	replicas := int32(2)
```
Comment (axsaucedo) [inline operator/controllers/seldondeployment_explainers_test.go:159]: Would it be worth adding also a test that checks for the explainer to be set on 0 replicas when the predictor has 0 even if explainer has 1+?

**Comment by axsaucedo** [comment]
Added a suggestion on test, but other than that looks good

File: operator/controllers/seldondeployment_explainers_test.go (hunk 1 above)
Comment (cliveseldon) [reply operator/controllers/seldondeployment_explainers_test.go]: added a test

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 02479ff7c9858feb0589094a514694313a2ff8d8 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-6clkp) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 02479ff7c9858feb0589094a514694313a2ff8d8 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-6clkp) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 02479ff7c9858feb0589094a514694313a2ff8d8 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-6clkp) | `/test notebooks`
integration | 02479ff7c9858feb0589094a514694313a2ff8d8 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-kk5zd) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 02479ff7c9858feb0589094a514694313a2ff8d8 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-6clkp) | `/test notebooks`
integration | 02479ff7c9858feb0589094a514694313a2ff8d8 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-kk5zd) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 02479ff7c9858feb0589094a514694313a2ff8d8 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-6clkp) | `/test notebooks`
integration | 1da64ac77e97a18cfd28a6f2ee017c614ce3184f | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-h8f4f) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 1da64ac77e97a18cfd28a6f2ee017c614ce3184f | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-h8f4f) | `/test integration`
notebooks | 1da64ac77e97a18cfd28a6f2ee017c614ce3184f | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-2kvtq) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 1da64ac77e97a18cfd28a6f2ee017c614ce3184f | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-h8f4f) | `/test integration`
notebooks | 1da64ac77e97a18cfd28a6f2ee017c614ce3184f | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-2kvtq) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 1da64ac77e97a18cfd28a6f2ee017c614ce3184f | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-2kvtq) | `/test notebooks`
integration | 5a13a928148db7af494f0f3308ab43886305d8bc | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-dglbx) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 1da64ac77e97a18cfd28a6f2ee017c614ce3184f | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-2kvtq) | `/test notebooks`
integration | 5a13a928148db7af494f0f3308ab43886305d8bc | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-x7hpz) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 5a13a928148db7af494f0f3308ab43886305d8bc | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-85tmt) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 5a13a928148db7af494f0f3308ab43886305d8bc | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-4lqzh) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by axsaucedo** [comment]
Merge conflict needs to be resolved to rerun tests

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | a109caa20aaa691d6fa1dbff6c16944bf35fa244 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-wd8lb) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by axsaucedo** [comment]
Notebooks and integration tests pass, and all comments resolved, merging
image

**Pull request merged**